service/service.go

```diff
@@ -103,6 +103,13 @@ func New(tarball io.Reader, env map[string]string, options ...Option) (*Service,
 	}
 	defer os.RemoveAll(s.tempPath)
 
+	// calculate hash
```

Comment:
It won't be deterministic here because tarball could be (un)compressed with diffrent algorithms.

Also tarball may have `.git` directory which will be removed when deploying.

Reply:
we are controlling the uncompression, if it's different then it's a different service. Same as the .git (but I agree it would be better to remove the .git when we create the tar)

Reply:
Forcing on one compression & archive type will limit the flexibility and we'll not be able to accept for example _.zip_ formats but at the of the day I think **we need this limitation**.

Because, in the marketplace we'll store service's hash in smart contracts. So services will not be there with their source code & hash together. When devs want to do their hash comparasation once they download a service, they can simply tar.gz services, calculate a sha1 and compare with the one on the marketplace. So this is very useful on security side and doesn't require devs to have special tools.

Reply:
The goal is to have the same hash when the service is well done (ignore .git, .DS_Store, dependency with tag or commit hash, etc...) and when the source is the same.
This is a good improvement over the version based on docker image, but it cannot be perfect because there are so many variable in place..